src/main/java/org/dimdev/dimdoors/pockets/VirtualDepthDependentPocket.java

```diff
@@ -0,0 +1,82 @@
+package org.dimdev.dimdoors.pockets;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.util.Pair;
+import org.dimdev.dimdoors.util.PocketGenerationParameters;
+import org.dimdev.dimdoors.world.pocket.Pocket;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class VirtualDepthDependentPocket extends VirtualPocket{
+	public static final String KEY = "depth_dependent";
+
+	private static final Codec<Pair<String, VirtualPocket>> PAIR_CODEC = RecordCodecBuilder.create(instance -> instance.group(
+			Codec.STRING.fieldOf("regex").forGetter(Pair::getLeft),
+			VirtualPocket.CODEC.fieldOf("pocket").forGetter(Pair::getRight)
+	).apply(instance, Pair::new));
+
+	public static final Codec<VirtualDepthDependentPocket> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+			Codec.STRING.fieldOf("id").forGetter(VirtualDepthDependentPocket::getName),
+			PAIR_CODEC.listOf().fieldOf("pockets").forGetter(VirtualDepthDependentPocket::getPocketList)
+	).apply(instance, VirtualDepthDependentPocket::new));
+
+
+
+	private final String name;
+	private final List<Pair<String, VirtualPocket>> pocketList;
```

Comment:
why not just a Map<String, VirtualPocket>

Reply:
it needs to retain the same order, not sure if a map would guarantee that

Reply:
LinkedHashMap?